src/main/java/codeu/model/store/basic/UserStore.java

```diff
@@ -121,4 +121,42 @@ public boolean isUserRegistered(String username) {
   public void setUsers(List<User> users) {
     this.users = users;
   }
+
+  /** Returns the number of users in UserStore */
+  public int numUsers() {
+    if (users != null) {
+      return users.size();
+    }
+    return 0;
+  }
+
+  /** Returns the last user added */
+  public User getLastUser() {
+    if (!users.isEmpty()) {
+      return users.get(users.size() - 1);
+    }
+    return null;
+  }
+
+  /** Returns the user with flipped is_admin flag */
+  public User setIsAdmin(User user, boolean is_admin) {
+    User newUser =
+        new User(
+            user.getId(),
+            user.getName(),
+            user.getHashedPassword(),
+            user.getCreationTime(),
+            is_admin);
+    persistentStorageAgent.writeThrough(newUser);
+
+    // update database with new user
+    for (int i = 0; i < users.size(); i++) {
```

Comment:
Maybe instead of trying to keep this list up to date all the time (potential race conditions here) - we can try directly writing and reading from the Datastore all the time :)

Reply:
I replaced the forloop with users = persistentStorageAgent.loadUsers() for now. Are you suggesting that everytime we call the local list users in Userstore, we do direct read and write?